IOSExercises/Views/DetailsViewController.m

```diff
@@ -0,0 +1,341 @@
+    }
+    
+    self.popoverViewController.view = [[UIView alloc] init];
+    [self.popoverViewController setPreferredContentSize:CGSizeMake(250.f, 270.f)];
+
+    //[self setBirthday];
+    if (!self.datePicker) {
+        self.datePicker = [[UIDatePicker alloc] init];
+    }
+    [self.datePicker setTranslatesAutoresizingMaskIntoConstraints:NO];
+    [self.datePicker setDatePickerMode:UIDatePickerModeDate];
+    [self.popoverViewController.view addSubview:self.datePicker];
+
+    UIButton *saveButton = [UIButton buttonWithType:UIButtonTypeSystem];
+    [saveButton setTranslatesAutoresizingMaskIntoConstraints:NO];
+    [saveButton setTitle:@"Save"
+                forState:UIControlStateNormal];
+    [saveButton addTarget:self
+                   action:@selector(saveBirthday:)
+         forControlEvents:UIControlEventTouchUpInside];
+
+    [self.popoverViewController.view addSubview:saveButton];
+
+    [self.popoverViewController setModalPresentationStyle:UIModalPresentationPopover];
+    [self presentViewController:self.popoverViewController
+                       animated:YES
+                     completion:nil];
+
+    NSDictionary *views = @{
+        @"datePicker" : self.datePicker,
+        @"saveButton" : saveButton
+    };
+
+    for (NSString *viewName in views) {
+        [NSLayoutConstraint activateConstraints:[NSLayoutConstraint constraintsWithVisualFormat:[NSString stringWithFormat:@"H:|[%@]|", viewName]
+                                                                                        options:0
+                                                                                        metrics:nil
+                                                                                          views:views]];
+    }
+
+    [NSLayoutConstraint activateConstraints:[NSLayoutConstraint constraintsWithVisualFormat:@"H:|[datePicker]|"
+                                                                                    options:0
+                                                                                    metrics:nil
+                                                                                    views:views]];
+
+    [NSLayoutConstraint activateConstraints:[NSLayoutConstraint constraintsWithVisualFormat:@"V:|-[datePicker]-[saveButton]"
+                                                                                    options:0
+                                                                                    metrics:nil
+                                                                                      views:views]];
+
+    UIPopoverPresentationController *popoverController = [self.popoverViewController popoverPresentationController];
+
+    [popoverController setPermittedArrowDirections:UIPopoverArrowDirectionAny];
+
+    [popoverController setSourceView:self.birthdayLabel];
+    [popoverController setSourceRect:self.birthdayLabel.bounds];
+}
+
+- (void)popoverWithGreeting:(NSString *)aGreeting {
+
```

Comment:
Have care in adding extra (empty) lines in code. It isn't a concern now, but once the class starts to bloat (due to increasing features), the extra lines mean less lines in the IDE's view (so make sure you really want them, eg for line grouping in terms of logic, param readability, and so on)